diff --git a/lib/es/MdEditor.mjs b/lib/es/MdEditor.mjs
index 7672a15a2fbf0a6af739c513371991f3199db562..005704cea8b6f3adc4733ffe0c8cb33bf2afe4cf 100644
--- a/lib/es/MdEditor.mjs
+++ b/lib/es/MdEditor.mjs
@@ -2992,6 +2992,12 @@ const Editor = /* @__PURE__ */ defineComponent({
     const state = reactive({
       scrollAuto: props2.scrollAuto
     });
+    watch(
+      () => props2.scrollAuto,
+      (v) => {
+        state.scrollAuto = v;
+      }
+    );
     const rootRef = ref();
     const codeRef = ref();
     useOnSave(props2, ctx);
diff --git a/lib/es/chunks/index.mjs b/lib/es/chunks/index.mjs
index 0238f7a3fe4bf0fb0ce895ed965086955e039b6f..d091c0eb1aed67d824658bdcddfea4408422f853 100644
--- a/lib/es/chunks/index.mjs
+++ b/lib/es/chunks/index.mjs
@@ -60,6 +60,9 @@ const useCopyCode = (props, html, key) => {
   const rootRef = inject("rootRef");
   const ult = inject("usedLanguageText");
   const initCopyEntry = () => {
+    if (!rootRef.value) {
+      return;
+    }
     rootRef.value.querySelectorAll(`#${editorId} .${prefix}-preview .${prefix}-code`).forEach((codeBlock) => {
       let clearTimer = -1;
       const copyButton = codeBlock.querySelector(
@@ -210,7 +213,7 @@ const useMermaid = (props) => {
     }
   });
   const replaceMermaid = async () => {
-    if (!props.noMermaid && mermaidRef.value) {
+    if (!props.noMermaid && mermaidRef.value && rootRef.value) {
       const mermaidSourceEles = rootRef.value.querySelectorAll(
         `div.${prefix}-mermaid`
       );
@@ -241,21 +244,23 @@ const useMermaid = (props) => {
                   svgContainingElement
                 );
                 mermaidHtml = await props.sanitizeMermaid(result.svg);
-                const p = document.createElement("p");
-                p.className = `${prefix}-mermaid`;
-                p.setAttribute("data-processed", "");
-                p.innerHTML = mermaidHtml;
-                (_a = p.children[0]) == null ? void 0 : _a.removeAttribute("height");
-                mermaidCache.set(item.innerText, p.innerHTML);
-                if (item.dataset.line !== void 0) {
-                  p.dataset.line = item.dataset.line;
-                }
-                item.replaceWith(p);
               } catch {
               }
-              if (--count === 0) {
-                svgContainingElement.remove();
+            }
+            if (mermaidHtml) {
+              const p = document.createElement("p");
+              p.className = `${prefix}-mermaid`;
+              p.setAttribute("data-processed", "");
+              p.innerHTML = mermaidHtml;
+              (_a = p.children[0]) == null ? void 0 : _a.removeAttribute("height");
+              mermaidCache.set(item.innerText, p.innerHTML);
+              if (item.dataset.line !== void 0) {
+                p.dataset.line = item.dataset.line;
               }
+              item.replaceWith(p);
+            }
+            if (--count === 0) {
+              svgContainingElement.remove();
             }
           };
           return handler(ele);
@@ -2268,9 +2273,7 @@ const useMarkdownIt = (props, previewOnly) => {
     props.onGetCatalog(headsRef.value);
     bus.emit(editorId, CATALOG_CHANGED, headsRef.value);
     nextTick(() => {
-      replaceMermaid().then(() => {
-        zoomMermaid(rootRef.value.querySelectorAll(`#${editorId} .${prefix}-mermaid`));
-      });
+      replaceMermaid();
     });
   };
   onMounted(updatedTodo);
@@ -2291,11 +2294,7 @@ const useMarkdownIt = (props, previewOnly) => {
     () => {
       if (props.setting.preview) {
         nextTick(() => {
-          replaceMermaid().then(() => {
-            zoomMermaid(
-              rootRef.value.querySelectorAll(`#${editorId} .${prefix}-mermaid`)
-            );
-          });
+          replaceMermaid();
           bus.emit(editorId, CATALOG_CHANGED, headsRef.value);
         });
       }
